### DISCUSSION TITLE
How can I force update a controller input component?

### DISCUSSION THREAD

**Discussion by c0d3x** [react-hook-form/react-hook-form#2965] [Q&A]
So I have a controlled input using the NumberFormat 3rd party lib to show currency.
Then I do some math from an array of values and define it as a totalAmount const, so I got the number.

How do I trigger an update of a readOnly controlled components value when I update a field in array to use the totalAmount value?

**Comment by bluebill1049** [comment]
`setValue`?

**Reply by c0d3x** [reply]
Nice, but what event can I use to trigger the setValue, it should set total amount based on a dynamic amount field where you can add more amounts.. So is there an event for each update of form?

**Reply by bluebill1049** [reply]
what do you mean `event` to update? i thought you only need the name of that input.

**Reply by c0d3x** [reply]
I tried to use setValue on the submit just to test it, it does not update the value of the controlled input that use the NumberFormat component. It has a onValueChange event that uses the target.floatValue from value. But it does update the form submit value, just not the html input value.

**Reply by bluebill1049** [reply]
are you using Contorller with `render`?

**Reply by c0d3x** [reply]
![image](https://user-images.githubusercontent.com/661911/93832606-84a57d80-fc76-11ea-8908-299a287e85bb.png)
Here it's easier to understand what I try to do.

**Reply by c0d3x** [reply]
My total value is not updating when I setValue.. the form itself updates just not the input that should be visible to user.

**Reply by c0d3x** [reply]
I am using controller like in this example:
https://codesandbox.io/s/react-hook-form-normalize-field-1jumn
